### PR TITLE
update recommedation page

### PR DESCRIPTION
**Pull request by Niko-Yuan**
1) add follower/post block
2) change UI/UX and make card more responsive
3) add fetch function to fetch user info from server.js
4) make the multiple Apikey changeable for deployment
5) add loading gif when waiting for fetching
6) add user image which is predefined
7) add backToTop button
8) upload several docs
9) add script in package.json to start all the servers
10) add profile page prototype

### CODE REVIEW

**Review by XSiling — COMMENTED** [review]
the bug for no user info return perhaps due to the fetching latency from the server.

the raw user info anchor can be found in the url(username)

**Comment by DevChuriwala** [comment]
@Niko-Yuan i tried running the tests on my local machine and it gave the same error. here is how i fixed it,

1. you can change the port to 3001, everywhere in the code.
2. you can use `lsof -i :3000` to see what is running on 3000 and kill it using `kill -9 <PID>`. after this is always works.

I am not sure why this is happening, and there is no way to do (2.) on github actions, i dont think
@friidryce any ideas?

**Comment by friidryce** [comment]
> @Niko-Yuan i tried running the tests on my local machine and it gave the same error. here is how i fixed it,
> 
> 1. you can change the port to 3001, everywhere in the code.
> 2. you can use `lsof -i :3000` to see what is running on 3000 and kill it using `kill -9 <PID>`. after this is always works.
> 
> I am not sure why this is happening, and there is no way to do (2.) on github actions, i dont think @friidryce any ideas?

I had the same issue on local tests before but they were working on GitHub actions

File: docs/fronted/api.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+### API Keys Module Documentation
```
Comment (friidryce) [inline docs/fronted/api.md:1]: Very nice

File: src/components/account.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  avatar.src = recommendationData.avatar;
+  avatar.alt = 'Avatar';
+  avatar.className = 'follower-card-avatar';
+
+  const nameUsernameContainer = document.createElement('section');
+  const followersFollowingContainer = document.createElement('section');
+  nameUsernameContainer.className = 'follower-card-name-username-container';
+  followersFollowingContainer.className = 'follower-card-followers-following-container';
+
+  const name = document.createElement('p');
```
Comment (friidryce) [inline src/components/account.js:27]: may need a little bit documentation on structure (for css editing)

File: src/components/recommendation.js [hunk 3]
```diff
@@ -1,19 +1,82 @@
...
+import { renderFollowerRecommendation } from "./account.js";
+import { flaskApikey, nodeApikey } from '../api/api.js';
 
-async function fetchRecommendations(interests) {
-  fetch("http://localhost:5000/get_recommendations")
+// TODO
+// 1) Merge fetchFollower and fetchPost into one promise
+// 2) add backToTop button
+
+function showLoadingGif(containerId) {
```
Comment (friidryce) [inline src/components/recommendation.js:10]: this seems like a good addition so it doesn't seem that the page is frozen

File: src/script.js [hunk 4]
```diff
@@ -1,146 +1,241 @@
...
-    ["Digital","Data", "Innovation","Gadgets","Software", "Internet", "Cybersecurity", "Programming"],
-    ["Finance", "Business", "Investment", "Banking", "Markets", "Stocks", "Wealth", "Budgeting"],
-    ["Home", "Pet", "Family", "Domestic", "Decor", "Garden", "Housing", "Comfort"],
-    ["Book", "School", "Library", "Learning", "Knowledge", "Study", "Research", "Literature"],
-    ["Sport", "Athletics", "Exercise", "Fitness", "Competition", "Games", "Outdoor", "Adventure"],
-    ["Emotion", "Relationship", "Charity", "Love", "Empathy", "Advocacy", "Philanthropy", "Volunteer"],
-]
+  ["Star", "Fun", "Movie", "TV", "Photography", "Music", "Pop", "Comic"],
+  [
+    "Beauty",
```
Comment (friidryce) [inline src/script.js:6]: Great update on readability

**Review by friidryce — APPROVED** [review]
Need these changes merged ASAP so I can also begin working on UI improvements, tests should pass now given my hotfix on ports

**Pull request merged**